### ISSUE TITLE
Jira login produces a "ERROR incorrect function"

### ISSUE THREAD

**Issue by SR-G** [Netflix-Skunkworks/go-jira#118]
Hey.
Seems like i have a login issue, i really don't see why.
JIRA server is version v7.1.4#71008

My command : `jira login -v` (-v does not add any additionnal informations here)
Result : `? Jira Password [myusername]:  ERROR Incorrect function.`
(don't ask for any password)

My config is : 
`
endpoint: http://myadress:8080
user: myusername
`

If i'm using (just an test in order to check myadress) from a browser the http://myadress:8080/rest/auth/1/session url i correctly get : 
`
{"self":"http://myadress:8080/rest/api/latest/user?username=myusername","name":"myusername","loginInfo":{"failedLoginCount":22,"loginCount":153,"lastFailedLoginTime":"2017-07-20T10:14:13.773+0200","previousLoginTime":"2017-10-03T11:30:07.112+0200"}}
`

Any idea ?

**Comment by coryb** [comment]
Hmm, no idea where `Incorrect function` is coming from, not my code at least. I think it is coming from the Jira server, perhaps a configuration issue?  With your browser you are just doing a GET to that endpoint, but that API requires you do to a POST to create a new session.

Here is a jira command to explicitly make that request and get full verbose output:
```
jira req -M POST -vvv /rest/auth/1/session '{"username":"myusername","password":"mypassword"}'
```

Maybe there is something in the response data that indicates what `Incorrect function` means, or perhaps talk to your Jira admins to see if they have any ideas why that API is returning `Incorrect function`?

**Comment by SR-G** [comment]
Ok so here is the very weird part. Seems the URL is not correct. Have to say that i'm using jira.exe (amd64) from Git Bash shell. Seems a variable is injected somewhere, i don't know why, how is the final URL built inside the code ?

> $ jira req -M POST -vvv /rest/auth/1/session '{"username":"myusername","password":"mypassword"}'
> DEBUG POST http://myadress:8080C:/Tools/PortableGit/2.9.2/rest/auth/1/session
> DEBUG Request: POST /Tools/PortableGit/2.9.2/rest/auth/1/session HTTP/1.1
> Host: myaddress:8080C:
> Accept: application/json
> Content-Type: application/json

You can guess that "C:/Tools/PortableGit/2.9.2/ is my where my bash exe is. I just have a "EXEPATH" variable (in `env`) pointing to that location. And my ~/.jira.d/config.xml file is clean (has just the adress, nothing more, excepted that the adress has :8080 at the end)

And i can confirm that it works without any problem under a plain cmd.exe (but which is way less convenient to use ...).

**Comment by coryb** [comment]
That is pretty strange, maybe it is bash on windows that is converting the `/rest/...` to a full path before sending it to `jira.exe`?  Maybe try adding quotes to that argument?  Not sure how this works on windows.  Given that it works under `cmd.exe` I would think it is a bash issue, pretty sure we dont have any prepend logic inside `jira` that would do this. 

You might get more details with `jira -vvv login` also which will construct the URL without bash involved.

**Comment by SR-G** [comment]
Yes, pretty sure you are right : this is probably added by the git bash itself, and only for the `jira req` command.

So right now : 
- `jira login` works each time from `cmd.exe`
- `jira login -vvv` still does not work from git bash, and i have the following output (see below). In fact, i think the problem is probably just about the way the password is read from STDIN, maybe this is what is producing the "incorrect function" under git bash. Then the jira server can't auth as no password have been provided. Have to say that other commands are able to read from STDIN from git bash without this error.

```
$ jira login -vvv
DEBUG GET http://myaddress:8080/rest/auth/1/session
DEBUG Request: GET /rest/auth/1/session HTTP/1.1
Host: myaddress:8080
Accept: application/json
Content-Type: application/json

DEBUG Cookie: atlassian.xsrf.token=BU29-XI95(...)|77eb7a6f7d989612a2840fe6fc86b35634c24c22|lin; JSESSIONID=46501A97556C(...)
DEBUG Response: HTTP/1.1 401 Unauthorized
Transfer-Encoding: chunked
Cache-Control: no-cache, no-store, no-transform
Content-Type: application/json;charset=UTF-8
Date: Wed, 04 Oct 2017 06:57:40 GMT
Server: Apache-Coyote/1.1
Set-Cookie: atlassian.xsrf.token=BU29-XI95-7FBH-VI68|825c3bb7eaac(...)|lout; Path=/
Vary: User-Agent
Www-Authenticate: OAuth realm="http%3A%2F%2Fmyaddress%3A8080"
X-Arequestid: 537x1072815x1
X-Asen: SEN-7330332
X-Ausername: anonymous
X-Content-Type-Options: nosniff

6f
{"errorMessages":["You are not authenticated. Authentication required to perform this operation."],"errors":{}}
0

DEBUG Setting DOMAIN to myaddress for Cookie: atlassian.xsrf.token=BU29-XI9(...)|825c3bb7eaac45c9ec83(...)|lout; Path=/
DEBUG Loading config C:\Users\myusername\.jira.d\config.yml
DEBUG Loading config C:\Users\myusername\.jira.d\config.yml
DEBUG GET http://myaddress:8080/rest/auth/1/session
DEBUG Request: GET /rest/auth/1/session HTTP/1.1
Host: myaddress:8080
Accept: application/json
Content-Type: application/json

DEBUG Cookie: atlassian.xsrf.token=BU29-XI95-7FBH-VI68|825c3bb7(...)|lout; JSESSIONID=46501A97556CE70D(...)
DEBUG Response: HTTP/1.1 401 Unauthorized
Transfer-Encoding: chunked
Cache-Control: no-cache, no-store, no-transform
Content-Type: application/json;charset=UTF-8
Date: Wed, 04 Oct 2017 06:57:40 GMT
Server: Apache-Coyote/1.1
Vary: User-Agent
Www-Authenticate: OAuth realm="http%3A%2F%2Fmyaddress%3A8080"
X-Arequestid: 537x1072816x1
X-Asen: SEN-7330332
X-Ausername: anonymous
X-Content-Type-Options: nosniff

6f
{"errorMessages":["You are not authenticated. Authentication required to perform this operation."],"errors":{}}
0

? Jira Password [myusername]:  ERROR Incorrect function.
```

**Comment by coryb** [comment]
Oh, interesting, it looks like it is not getting passed the password prompt.  Sorry I was confused and though the error was coming from Jira, but it must be coming from the password prompting library I am using.  This one will be tricky to fix.  So it works for you under `cmd.exe` and does not work when running under `bash.exe`?  Are you running Cygwin bash or is this via the Linux Subsystem in Windows?  Trying to figure out how I can preproduce this to try to fix.

**Comment by SR-G** [comment]
Yes : works under "cmd.exe" but not under "bash.exe". I'm using the shell provided by the Windows GiT installation, as far as i know it's based on MINGW. You can just grab it here : https://git-scm.com/download/win (portable installation for example), then launch `git-bash.exe`.
